internal/pkg/gateway/registry.go

```diff
@@ -348,7 +346,41 @@ func (reg *registry) registerChannel(channel string) error {
 			}
 		}
 	}
-	reg.channelsInitialized[channel] = true
+	reg.channelInitialized[channel] = true
 
 	return nil
 }
+
+func (reg *registry) updateConfig(channel string) ([]*endpointConfig, error) {
+	config, err := reg.discovery.Config(channel)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get config for channel [%s]: %w", channel, err)
+	}
+	// get the tlscerts
+	for msp, info := range config.GetMsps() {
+		reg.tlsRootCerts.Store(msp, info.GetTlsRootCerts())
```

Comment:
I don't understand. Why are we aggregating all TLS root CAs? Discovery, and also the channel config, separates TLS CAs by their MSP, and here you are aggregating them together.

This means that a peer issued by the CA of organization A can be impersonated by a rogue admin of organization B.

Reply:
This cache is not needed anymore - I've removed it